pkg/pool-linear/contracts/test/MockStaticAToken.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-standalone-utils/contracts/test/TestToken.sol";
+
+import "../interfaces/IStaticAToken.sol";
+
+contract MockStaticAToken is TestToken, IStaticAToken, ILendingPool {
+    uint256 private _rate = 1e27;
+    address private immutable _ASSET;
+
+    constructor(
+        address admin,
+        string memory name,
+        string memory symbol,
+        uint8 decimals,
+        address underlyingAsset
+    ) TestToken(admin, name, symbol, decimals) {
+        _ASSET = underlyingAsset;
+    }
+
+    // solhint-disable-next-line func-name-mixedcase
+    function ASSET() external view override returns (address) {
+        return _ASSET;
+    }
+
+    // solhint-disable-next-line func-name-mixedcase
+    function LENDING_POOL() external view override returns (ILendingPool) {
+        return ILendingPool(this);
```

Comment:
:joy:

Reply:
This is totally fine as a solution, the alternative approach I would've gone with would be to deploy the lending pool in this contract's constructor.